app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -6,15 +6,15 @@ function checkIfNearBottomOfPage() {
   var elCheck = document.getElementById("index-container");
```

Comment:
Sure. I'm going to resolve the conflicts I got on this file, and then I'm going to see what I can refactor to remove the linting errors.

Reply:
@rhymes, I fixed the errors and warnings in this file (they're here https://gist.github.com/johncurcio/8de6a546ebc0a1cd6c4ed0a1fb67bbfc), however I feel like a couple of linting rules made the file harder to read or are unnecessary:

1. **no-use-before-define**: ``initScrolling`` is kind of the main function in this file, and I think it should be at the top, but because it indirectly calls ``fetchNextPageIfNearBottom`` which calls most functions in the file, ``initScrolling`` has to be placed at the bottom. I think that makes it harder to understand where to start reading the functions.

2. **no-console**: is it a problem to have the console logging an error?

I think we could ignore a few of these linting rules, unless there's a better way to do this that I'm not seeing. What do you think?

Reply:
I understand your point but I think it's fine, it follows a pattern (maybe not an ideal one) but one that's consistent in all vanilla JS file we have. When we'll be able to move that code over to webpack and use ES6 modules it'll be clearer what's the entry point :)

Regarding the console error I think it's another thing we might have to live with right now, unless we actually implement a better error handling there, it's probably better to log the error instead of hiding 👀

Reply:
I also think we should add `use strict` to this file